kernel/sched.c

```diff
@@ -781,13 +781,10 @@ void _impl_k_yield(void)
 Z_SYSCALL_HANDLER0_SIMPLE_VOID(k_yield);
 #endif
 
-void _impl_k_sleep(s32_t duration)
+s32_t _impl_k_sleep(s32_t duration)
```

Comment:
`u32_t` unless some sort of error can be return too (this is not documented)

Reply:
Yes that was my assumption. Returning -Exxx

Reply:
In Zephyr time is represented as s32_t. Return value follows this convention.